File: master/models/fleet.go
package models

import "gorm.io/gorm"

type Fleet struct {
	gorm.Model
	LightFighter int  `json:"lf"`
	HeavyFighter int  `json:"hf"`
	Cruiser      int  `json:"cr"`
	Battleship   int  `json:"bs"`
	Dreadnought  int  `json:"dr"`
	Destroyer    int  `json:"de"`
	Deathstar    int  `json:"ds"`
	Bomber       int  `json:"bomb"`
	Guardian     int  `json:"guard"`
	Satellite    int  `json:"satellite"`
	Cargo        int  `json:"cargo"`
	TaskID       uint `json:"task_id"`
}

type FleetDTO struct {
	LightFighter int `json:"lf"`
	HeavyFighter int `json:"hf"`
	Cruiser      int `json:"cr"`
	Battleship   int `json:"bs"`
	Dreadnought  int `json:"dr"`
	Destroyer    int `json:"de"`
	Deathstar    int `json:"ds"`
	Bomber       int `json:"bomb"`
	Guardian     int `json:"guard"`
	Satellite    int `json:"satellite"`
	Cargo        int `json:"cargo"`
}

func (fleet Fleet) ToDTO() *FleetDTO {
	return &FleetDTO{
		LightFighter: fleet.LightFighter,
		HeavyFighter: fleet.HeavyFighter,
		Cruiser:      fleet.Cruiser,
		Battleship:   fleet.Battleship,
		Dreadnought:  fleet.Dreadnought,
		Destroyer:    fleet.Destroyer,
		Deathstar:    fleet.Deathstar,
		Bomber:       fleet.Bomber,
		Guardian:     fleet.Guardian,
		Satellite:    fleet.Satellite,
		Cargo:        fleet.Cargo,
	}
}
func (fleet Fleet) GetEntityPrefix() string {
	return "fleet_"
}

File: node/model/fleet.py
from dataclasses_json import dataclass_json
from dataclasses import dataclass
from config import ShipToID


@dataclass_json
@dataclass
class Fleet:
    ds: int = 0       # death star
    de: int = 0       # destroyer
    cargo: int = 0    # cargo ship
    bs: int = 0       # battleship
    satellite: int = 0  # satellite
    lf: int = 0       # light fighter
    hf: int = 0       # heavy fighter
    cr: int = 0       # cruiser
    dr: int = 0       # dreadnought
    bomb: int = 0     # bomber
    guard: int = 0    # guard ship

    # Method 1: Using __dict__ and dict comprehension
    def to_fleet(self):
        return {ShipToID[attr]: value for attr, value in self.__dict__.items()}


if __name__ == '__main__':
    pass
